From 02ab6686123c14029bf316da1ccb7d6e3e066df9 Mon Sep 17 00:00:00 2001
From: Pekka Paalanen <pekka.paalanen@collabora.com>
Date: Thu, 4 Apr 2019 17:11:53 +0300
Subject: [PATCH 14/45] libweston: export
 weston_linux_sync_file_read_timestamp()

This is an internal export for GL-renderer, so that it does not need to build
linux-sync-file.c a second time. This follows the example of
linux-explicit-synchronization.c which is also used by GL-renderer.

Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>
---
 libweston/gl-renderer.c     | 3 ++-
 libweston/linux-sync-file.c | 5 +++--
 libweston/linux-sync-file.h | 2 +-
 libweston/meson.build       | 1 -
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/libweston/gl-renderer.c b/libweston/gl-renderer.c
index a03e45ea..d160a5d8 100644
--- a/libweston/gl-renderer.c
+++ b/libweston/gl-renderer.c
@@ -339,7 +339,8 @@ timeline_render_point_handler(int fd, uint32_t mask, void *data)
 	if (mask & WL_EVENT_READABLE) {
 		struct timespec tspec = { 0 };
 
-		if (linux_sync_file_read_timestamp(trp->fd, &tspec) == 0) {
+		if (weston_linux_sync_file_read_timestamp(trp->fd,
+							  &tspec) == 0) {
 			TL_POINT(tp_name, TLP_GPU(&tspec),
 				 TLP_OUTPUT(trp->output), TLP_END);
 		}
diff --git a/libweston/linux-sync-file.c b/libweston/linux-sync-file.c
index 913fb8da..9f5313cc 100644
--- a/libweston/linux-sync-file.c
+++ b/libweston/linux-sync-file.c
@@ -30,6 +30,7 @@
 #include <poll.h>
 #include <stddef.h>
 #include <sys/ioctl.h>
+#include <wayland-server-core.h>
 
 #ifdef HAVE_LINUX_SYNC_FILE_H
 #include <linux/sync_file.h>
@@ -62,8 +63,8 @@ linux_sync_file_is_valid(int fd)
  * \param ts[out] the timespec struct to fill with the timestamp
  * \return 0 if a timestamp was read, -1 on error
  */
-int
-linux_sync_file_read_timestamp(int fd, struct timespec *ts)
+WL_EXPORT int
+weston_linux_sync_file_read_timestamp(int fd, struct timespec *ts)
 {
 	struct sync_file_info file_info = { { 0 } };
 	struct sync_fence_info fence_info = { { 0 } };
diff --git a/libweston/linux-sync-file.h b/libweston/linux-sync-file.h
index b831fa1e..9746d7ba 100644
--- a/libweston/linux-sync-file.h
+++ b/libweston/linux-sync-file.h
@@ -33,6 +33,6 @@ bool
 linux_sync_file_is_valid(int fd);
 
 int
-linux_sync_file_read_timestamp(int fd, struct timespec *ts);
+weston_linux_sync_file_read_timestamp(int fd, struct timespec *ts);
 
 #endif /* WESTON_LINUX_SYNC_FILE_H */
diff --git a/libweston/meson.build b/libweston/meson.build
index d338c9ab..aeafcc5d 100644
--- a/libweston/meson.build
+++ b/libweston/meson.build
@@ -431,7 +431,6 @@ if get_option('renderer-gl')
 
 	srcs_renderer_gl = [
 		'gl-renderer.c',
-		'linux-sync-file.c',
 		'../shared/matrix.c',
 		linux_dmabuf_unstable_v1_protocol_c,
 		linux_dmabuf_unstable_v1_server_protocol_h,
-- 
2.21.0

